students/thorn/lesson04/trigram.py

```diff
@@ -28,91 +28,76 @@ def read_text(intext):
         # Strip to only letters and spaces between words, then add to list.
         formatted_text = re.sub(r'\W', ' ', raw_text.strip().lower())
         text_list = list(formatted_text.split())
-    return text_list            
-
-def create_digrams(text_list):
-    """
-    FOR TESTING ONLY
-
-    Creates a list of digram tuples from the text.  
-    
-    Digram consist of the first word and the word following.
-    Ex: ([0],[0+1])
-    """
-    digrams = []
-    for i in range(len(text_list)):
-        if i < len(text_list)-1:
-            digrams.append((text_list[i], text_list[i+1]))
-    return digrams    
+    return text_list
 
 
-def create_trigrams(text_list):
-    """
-    Creates a list of trigram tuples from the text.
-
-    Trigrams consist of the first 2 words and the word following.
-    Ex: ([0],[0+1],[0+2])
-    """
-    trigrams = []
+def create_trigrams_try_two(text_list):
+    trigrams = {}
+    # Add 3rd word to trigram pairs.  Create key entry if not already in dict.
     for i in range(len(text_list)-2):
-        trigrams.append((text_list[i], text_list[i+1], text_list[i+2]))
-
-    # Create set and shuffle for randomness.
-    random.shuffle(trigrams)
-
-    trigram_set = set(trigrams)
-    return trigram_set
+        try:
+            trigrams[(text_list[i], text_list[i+1])].append(text_list[i+2])
+        except:
+            trigrams[(text_list[i], text_list[i+1])] = (text_list[i+2])
```

Comment:
1. which exception are you catching? it is best to catch specific exception.
2. checkout `defaultdict` from collections module